scene/gui/button_array.cpp

```diff
@@ -320,6 +328,21 @@ void ButtonArray::_gui_input(const Ref<InputEvent> &p_event) {
 			if (ofs >= buttons[i]._pos_cache && ofs < buttons[i]._pos_cache + buttons[i]._size_cache) {
 
 				set_selected(i);
+				emit_signal("button_pressed", i);
+				return;
+			}
+		}
+	}
+
+	if (mb.is_valid() && mb->is_pressed() && mb->get_button_index() == BUTTON_LEFT) {
+
+		int ofs = orientation == HORIZONTAL ? mb->get_position().x : mb->get_position().y;
+
+		for (int i = 0; i < buttons.size(); i++) {
+
+			if (ofs >= buttons[i]._pos_cache && ofs < buttons[i]._pos_cache + buttons[i]._size_cache) {
+
```

Comment:
if I remove those, the signal won't be able to fire when the mouse triggers the event.(or is there a better implementation to this?)
I am fairly new to the c++ workflow, so excuse any stupid mistakes..